app/Recources/UpgradeBluePrint.gd

```diff
@@ -0,0 +1,19 @@
+extends Resource
+
+class_name Upgrades
+
+export var UpgradeName : String = ""
+export var UpgradeDescription : String = ""
+export var AddedHealthToBoidsAndPlayer : float
+export var AddedStaminaToPlayer : float
+export var AddedDamageToBoids : float
+export var AddedHealthToEnemies : float
+export  var AddedDamageToEnemies : float
+
+func Upgrade():
+	GlobalUpgradeStats.globalSelfHealth += AddedHealthToBoidsAndPlayer
+	GlobalUpgradeStats.playerStamina += AddedStaminaToPlayer
+	GlobalUpgradeStats.boidDamage += AddedDamageToBoids
+	GlobalUpgradeStats.globalEnemyHealth += AddedHealthToEnemies
+	GlobalUpgradeStats.globalEnemyDamage += AddedDamageToEnemies
+	PlayerStats.UpgradedValues()
```

Comment:
this should probably be in /Scripts

Reply:
should this  have an _init function? https://docs.godotengine.org/en/3.5/tutorials/scripting/resources.html#creating-your-own-resources

Reply:
Why _init ? I think it's nice to have it in resources because it's easier to make new instances of it, I could change it tho

Reply:
Yeah i dont see where you create a new instance though. It looks like the _init method is used for creating instances

Reply:
I think I was confused about how you were using the resource

Reply:
Oh that is just by hand
you click on the blueprint and click create new resource and you just fill in the blanks